components/script/dom/bluetoothremotegattcharacteristic.rs

```diff
@@ -140,16 +140,18 @@ impl BluetoothRemoteGATTCharacteristic {
             Ok(descriptor_vec) => {
                 let context = self.service.get().get_device().get_context();
                 for descriptor in descriptor_vec {
-                    if let Some(existing_descriptor) = context.get_descriptor_map()
-                                                              .borrow()
-                                                              .get(&descriptor.instance_id.clone()) {
-                        descriptors.push(existing_descriptor.get());
-                        continue;
-                    }
-                    let bt_descriptor = BluetoothRemoteGATTDescriptor::new(&self.global(),
-                                                                           self,
-                                                                           DOMString::from(descriptor.uuid),
-                                                                           descriptor.instance_id.clone());
+                    let bt_descriptor = match context.get_descriptor_map()
```

Comment:
How about you store the list in a var, so you don't have to fetch it every time?